text/0000-rust-analyzer.md

```diff
@@ -0,0 +1,155 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: 2020-04-20
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+- master [compiler team IDE strategy and planning](/9qrJYdSLTK2MWG4-pbmpWw?edit) document
+
+# Summary
+[summary]: #summary
+
+The RFC proposes a plan to adopt rust-analyzer as Rust's official LSP implementation. The transition to rust-analyzer will take place in a few stages:
+
+* **Feedback** -- encourage people to use rust-analyzer and report problems
+* **Deprecation** -- actively transition people from the RLS to rust-analyzer
+* **Transition** -- stop supporting the older RLS
+
+As detailed below, one major concern with rust-analyzer as it stands today is that it shares very little code with rustc. To avoid creating an unsustainable maintainance burden, this RFC proposes extracting shared libraries that will be used by both rustc and rust-analyzer ("library-ification"), which should eventually lead to rustc and rust-analyzer being two front-ends over a shared codebase.
```

Comment:
Please not “library-ification”. “librarification” or “librification”.

Reply:
While the original wording does seem a little silly  I perosnally find it way easier to read and understand than the suggested contracted versions.

Reply:
While the original wording does seem a little silly  I perosnally find it way easier to read and understand than the suggested contracted versions.
EDIT: Also, English in general is not really a language where you join words together like that, maybe "Library Extraction" serves the purpose better?

Reply:
Library-fication would be most [exact](https://www.yourdictionary.com/fication), if they intend to describe the goal.  
If they are doing it now its ying/y-ification.

Reply:
@kiljacken the whole is becoming a collection of libraries, not only a part

Reply:
Not sure how much we're invested in using the actual term but maybe _modularization_ would be better? (It does remind me of Rust modules, though...)

Reply:
Since Rust reads very nicely like a book, I prefer the book-related term. Arization comes from arize comes from arise which sounds like something pops up unexpected. ;-)

Reply:
lol, I am for some odd reason quite fond of the "obviously incorrect" library-ification spelling, and it's definitely what we say more commonly when speaking out loud and typing informally, in contrast to librarification. 

I don't like "modularization", even though I agree it is more correct, because it doesn't really communicate the "scale" to me.  (As you said, @Xanewok, it sounds like Rust modules.)

In any case, I.. don't really want to debate this particular terminology right here, as it is hardly central to the RFC. Maybe we can come up with some clever pun based on the term *crates* though...